gcloud/logging/logging_service_v2_api.py

```diff
@@ -0,0 +1,208 @@
+            ssl_creds=None,
+            scopes=_ALL_SCOPES,
+            is_idempotent_retrying=True,
+            max_attempts=3,
+            timeout=30):
+        self.defaults = api_callable.ApiCallableDefaults(
+            timeout=timeout,
+            max_attempts=max_attempts,
+            is_idempotent_retrying=is_idempotent_retrying)
+        self.stub = api_utils.create_stub(
+            logging_pb2.beta_create_LoggingServiceV2_stub,
+            service_path,
+            port,
+            ssl_creds=ssl_creds,
+            channel=channel,
+            scopes=scopes)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
+
+    def close(self):
+        del self.stub
+
+    # Service calls
+    def delete_log(
+            self,
+            log_name='',
+            **kwargs):
+        """
+        Deletes a log and all its log entries.
+        The log will reappear if it receives new entries.
+        """
+        delete_log_request = logging_pb2.DeleteLogRequest(
+            log_name=log_name,
+            **kwargs)
+        return self.delete_log_callable()(delete_log_request)
+
+    def delete_log_callable(
+            self,
+            is_retrying=None,
+            max_attempts=None):
+        return api_callable.idempotent_callable(
+            self.stub.DeleteLog,
+            is_retrying=is_retrying,
+            max_attempts=max_attempts,
+            defaults=self.defaults)
+
+    def write_log_entries(
+            self,
+            log_name='',
+            resource=None,
+            labels=None,
+            entries=None,
+            **kwargs):
+        """
+        Writes log entries to Cloud Logging.
+        All log entries in Cloud Logging are written by this method.
+        """
+        if resource is None:
+            resource = monitored_resource_pb2.MonitoredResource()
+        if labels is None:
+            labels = []
+        if entries is None:
+            entries = []
+        write_log_entries_request = logging_pb2.WriteLogEntriesRequest(
+            log_name=log_name,
+            resource=resource,
+            labels=labels,
+            entries=entries,
+            **kwargs)
+        return self.write_log_entries_callable()(write_log_entries_request)
+
+    def write_log_entries_callable(
+            self,
+            is_retrying=None,
+            max_attempts=None):
+        return api_callable.idempotent_callable(
+            self.stub.WriteLogEntries,
+            is_retrying=is_retrying,
+            max_attempts=max_attempts,
+            defaults=self.defaults)
+
+    def list_log_entries(
+            self,
+            project_ids=None,
+            filter_='',
+            order_by='',
+            **kwargs):
+        """
+        Lists log entries.  Use this method to retrieve log entries from Cloud
+        Logging.  For ways to export log entries, see
+        [Exporting Logs](/logging/docs/export).
```

Comment:
I want to fix this in the generator itself... it's a little difficult because this documentation is pulled from the proto.
I'm investigating how to do this properly but would rather hold off till the next PR to fix the docstrings.